src/integration/java/io/pinecone/integration/dataPlane/ListEndpointTest.java

```diff
@@ -0,0 +1,78 @@
+package io.pinecone.integration.dataPlane;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.pinecone.clients.AsyncIndex;
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.helpers.TestIndexResourcesManager;
+import io.pinecone.proto.ListResponse;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ListEndpointTest {
+    private static final TestIndexResourcesManager indexManager = TestIndexResourcesManager.getInstance();
+    private static String indexName ;
+    private static Index indexConnection;
+    private static AsyncIndex asyncIndexConnection;
+
+    @BeforeAll
+    public static void setUp() throws InterruptedException {
+        indexName = indexManager.getServerlessIndexName();
+        indexConnection = indexManager.getServerlessIndexConnection();
+        asyncIndexConnection = indexManager.getServerlessAsyncIndexConnection();
+    }
+
+    @AfterAll
+    public static void cleanUp() {
+        String apiKey = System.getenv("PINECONE_API_KEY");
+        Pinecone pinecone = new Pinecone.Builder(apiKey).build();
+        pinecone.deleteIndex(indexName);
```

Comment:
You do not need to explicitly delete the index here. It's meant to be shared across tests so having a single test file delete the index means the `TestIndexResourcesManager` will need to create it again for any additional tests, and wait to seed, which takes a lot of time.

Reply:
Ahhh okay. But we don't explicitly delete it in the `TestIndexResourcesManager` singleton, do we?

Reply:
Oh wait I think we can via the `cleanupResources` method... and I see now that it's used in the `CleanupAllTestResourcesListener` class. Nice!!